cmd/registry/cmd/compute/complexity.go

```diff
@@ -122,16 +128,22 @@ func (task *computeComplexityTask) Run(ctx context.Context) error {
 	} else {
 		return fmt.Errorf("we don't know how to summarize %s", task.specName)
 	}
-	subject := task.specName
-	messageData, _ := proto.Marshal(complexity)
-	artifact := &rpc.Artifact{
-		Name:     subject + "/artifacts/" + relation,
-		MimeType: core.MimeTypeForMessageType("gnostic.metrics.Complexity"),
-		Contents: messageData,
-	}
-	err = core.SetArtifact(ctx, task.client, artifact)
-	if err != nil {
-		return err
+
+	if task.dryRun {
+		core.PrintMessage(complexity)
+	} else {
```

Comment:
Nit: Prefer stopping the function early over indenting the non-dry-run code one level deeper.

```
if task.dryRun {
	core.PrintMessage(complexity)
	return nil
}

// non-dry-run code
```